src/current/v23.1/advanced-changefeed-configuration.md

```diff
@@ -33,6 +33,39 @@ kvadmission.rangefeed_catchup_scan_elastic_control.enabled
 
 For a more technical explanation of elastic CPU, refer to the [Rubbing control theory on the Go scheduler](https://www.cockroachlabs.com/blog/rubbing-control-theory/) blog post.
 
+### Improving latency
+
+Latency in changefeeds is the sum of the time it takes to:
+
+- Commit writes to the database.
+- Encode [changefeed messages]({% link {{ page.version.version }}/changefeed-messages.md %}).
+- Deliver the message to the [sink]({% link {{ page.version.version }}/changefeed-sinks.md %}).
+
+To improve changefeed latency, you can use the following [cluster settings]({% link {{ page.version.version }}/cluster-settings.md %}) to reduce bursts of rangefeed work so that updates are paced steadily over time. It is necessary to adjust these cluster settings together in order to have an effect on changefeed latency. However, note that these can have an effect on running transactions, and should be adjusted and tested with your particular workload.
+
+#### `kv.closed_timestamp_target_duration`
+
+Each [range]({% link {{ page.version.version }}/architecture/overview.md %}#range) in CockroachDB tracks a property called the [_closed timestamp_]({% link {{ page.version.version }}/architecture/transaction-layer.md %}#closed-timestamps), which means that no new writes can ever be introduced at or below that timestamp. Decreasing the target duration can reduce changefeed latency, so that the changefeed job can [checkpoint]({% link {{ page.version.version }}/how-does-an-enterprise-changefeed-work.md %}) more frequently and deliver messages to the sink.
+
```

Comment:
This section is mostly incorrect -- sorry, @kathancox .
It's not on you, it's just these
settings have been the source of great many problems in the past, and continue to be very problematic.

Recently, the default on master was bumped to 3s (from old value of 0 -- which then used the value of the setting below -- the 200ms `kv.closed_timestamp.side_transport_interval` setting).  We have some customers who run with even higher values.

My first recommendation is: don't touch this setting unless you know what you're doing (more on this below).

My second recommendation is to never set it low.  If this setting was ever altered by the customer, I recommend they reset this setting value to 3s or higher.

So, what does this setting do? it controls the frequency of checkpoints for each range.  Changefeed aggregates those checkpoints across *all* ranges, and once all ranges advance -- then the changefeed can checkpoint.  Therefore, the higher the value of this setting the longer it could take for changefeed to checkpoint.  It is important to understand that in its default configuration changefeed does not checkpoint more often than once every 30 seconds.    So, by setting this setting to be low, you're actually not making changefeed do anything other than adding more load on your cluster.

As mentioned above, some customers set this setting to be as high as 30s.  And the reason why you may want to do that is to attempt to lower the impact of changefeeds on SQL latency.  The reasons are rather deep and technical, but I think we can comfortably say that increasing this setting will reduce the load on the cluster and, conversely,  decreasing it will do exact opposite.
This is particularly important for very large scale customers, and I think we should call these out.  Basically, unless you're in TB range, there is no reason for you to touch this setting. If you have very large tables, consider *increasing* this setting.


I do not recommend it being set higher than say 20-30s.  In all honesty, the new default of 3 seconds is pretty good.

Next, we need to understand, and *fix* misunderstanding that somehow this setting impacts changefeed latency.  As I explained above, this setting *could* impact how often changefeed can checkpoints.  It could impact how often changefeed will emit "resolved" events.  But this does *not* impact how long it will take for changefeed to emit real event (an update).  I suspect the reason for this is the "max changefeed latency" graph we had for very long time where the time since the last checkpoint was treated as "latency" -- and let's be very clear: it is *not* event latency.

Reply:
That's OK, thanks for the detail. There wasn't much information on this around, so it is not surprising this needs some adjustment!

I have synthesized your explanation about this setting into the section, and not recommended any change unless customers have large workloads. I also adapted the intro to the overall section to mention how we're not just talking about changefeed event latency. 

Does these changes but reflect what we want here?